CRM/Core/BAO/CMSUser.php

```diff
@@ -40,9 +40,12 @@ public static function create(&$params, $mailParam) {
 
     $ufID = $config->userSystem->createUser($params, $mailParam);
 
-    //if contact doesn't already exist create UF Match
-    if ($ufID !== FALSE &&
-      isset($params['contactID'])
+    // Create UF Match if we have contactID unless we're Standalone
+    // since in Standalone uf_match is the same table as User.
+    if (
+      CIVICRM_UF !== 'Standalone'
+      && $ufID !== FALSE
+      && isset($params['contactID'])
```

Comment:
This seems to me like an small instance of a more general question.  Since this one is small, so it feels OK to throw in the quick conditional. But I just want to verbalize the general form to make sure we're good. The question is: how does this affect portability/DX/AX of files/screens/use-cases/callers that touch on user-information.

One random example:

* In the "View Contact" screen, it always annoys me that I can see "User ID" but **not** "Username". So maybe at some point we fix "View Contact" and we have "View Contact" pulling data via APIv4. (*as would happen with afform-based block*)
* tldr: You have a shareable resource (`*.aff.html`) that mentions either the word `UFMatch` or the word `User`. And some person (ext-dev, core-dev, site-builder) has to choose the word `UFMatch` or the word `User`.

From the MM discussion of "Plan B", the implication seems to be:

1. _You should usually choose `UFMatch` since that's more portable._
2. _You may sometimes choose `User` - this will make it non-portable and will grant access to more fields._

I'm certain that some asymmetry will necessarily be in the system - `Standalone` needs functionality that's not in the UF abstraction. (*OK, one could write bridges for each UF to support `User` API... but probably not us, probably not now*). I just to make sure we're aware/accepting of where the asymmetry lands.

Reply:
Or as tldr: "_`User` API lives in `standaloneusers`. You either declare a dependency on `standaloneusers`, or you don't use `User` API - so pick whatever else is available._"

Which is probably fair...